pkg/kubelet/kubelet_pods.go

```diff
@@ -1635,7 +1635,15 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 				}
 				// Secondary IP is not set #105320
 				if len(hostIPs) == 2 && len(s.PodIPs) == 1 {
-					s.PodIPs = append(s.PodIPs, v1.PodIP{IP: hostIPs[1].String()})
+					currHostIPIsIPv6 := utilnet.ParseIPSloppy(s.PodIPs[0].IP).To4() == nil
+					newHostIPIsIPv6 := utilnet.ParseIPSloppy(hostIPs[1].String()).To4() == nil
+					if currHostIPIsIPv6 != newHostIPIsIPv6 {
+						s.PodIPs = append(s.PodIPs, v1.PodIP{IP: hostIPs[1].String()})
+					} else {
+						kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidHostIPs",
+							"IP family of the second host IP %q is the same as the first one %q",
+							hostIPs[1].String(), s.PodIPs[0].IP)
```

Comment:
I think we should give some clue to the user, we know the cause that is that the node-ip flag and the order returned by the ccm doesn't match, I think that something like this will make it
```suggestion
							"Inconsistent addresses order received from the cloud-provider, IP family of the second host IP %q is the same as the first one configured in the kubelet %q",
							hostIPs[1].String(), s.PodIPs[0].IP)
```